src/init.cpp

```diff
@@ -1588,7 +1588,7 @@ bool AppInitMain(const util::Ref& context, NodeContext& node, interfaces::BlockA
                 // If the loaded chain has a wrong genesis, bail out immediately
                 // (we're likely using a testnet datadir, or the other way around).
                 if (!chainman.BlockIndex().empty() &&
-                        !LookupBlockIndex(chainparams.GetConsensus().hashGenesisBlock)) {
+                        !g_chainman.m_blockman.LookupBlockIndex(chainparams.GetConsensus().hashGenesisBlock)) {
```

Comment:
What is the point of moving a helper that accesses the global to be a member method of the class of the global, which is then called from the global again?

This should use the local reference (like the line immediately above it)

```suggestion
                        !chainman.m_blockman.LookupBlockIndex(chainparams.GetConsensus().hashGenesisBlock)) {
```

Feedback also applies to all other lines in this patch ;)

Reply:
```suggestion
                        !g_chainman.LookupBlockIndex(chainparams.GetConsensus().hashGenesisBlock)) {
```

Also, to avoid verbosity, chainman could forward the call to its blockman member.

Reply:
Continuing discussion here: https://github.com/bitcoin/bitcoin/issues/20049#issuecomment-702323178